x-pack/plugins/apm/server/lib/service_map/get_service_map_service_node_info.ts

```diff
@@ -173,25 +177,49 @@ async function getMemoryStats({
   filter,
 }: TaskParameters): Promise<{ avgMemoryUsage: number | null }> {
   const { apmEventClient } = setup;
-  const response = await apmEventClient.search({
-    apm: {
-      events: [ProcessorEvent.metric],
-    },
-    body: {
-      query: {
-        bool: {
-          filter: [
-            ...filter,
-            { exists: { field: METRIC_SYSTEM_FREE_MEMORY } },
-            { exists: { field: METRIC_SYSTEM_TOTAL_MEMORY } },
-          ],
+
+  const getAvgMemoryUsage = async ({
+    additionalFilters,
+    script,
+  }: {
+    additionalFilters: ESFilter[];
+    script: typeof percentCgroupMemoryUsedScript;
+  }) => {
+    const response = await apmEventClient.search({
+      apm: {
+        events: [ProcessorEvent.metric],
+      },
+      body: {
+        query: {
+          bool: {
+            filter: [...filter, ...additionalFilters],
+          },
+        },
+        aggs: {
+          avgMemoryUsage: { avg: { script } },
         },
       },
-      aggs: { avgMemoryUsage: { avg: { script: percentMemoryUsedScript } } },
-    },
-  });
+    });
 
-  return {
-    avgMemoryUsage: response.aggregations?.avgMemoryUsage.value ?? null,
+    return response.aggregations?.avgMemoryUsage.value ?? null;
   };
+
+  let avgMemoryUsage = await getAvgMemoryUsage({
+    additionalFilters: [
+      { exists: { field: METRIC_CGROUP_MEMORY_USAGE_BYTES } },
+    ],
+    script: percentCgroupMemoryUsedScript,
```

Comment:
Can we formulate this as a general question and ask someone from the elasticsearch team if this is the right approach?
It seems like a pretty common thing to do aggregations on fields that are not guaranteed to be in the mappings of older indices. Instead of crashing the query, those indices should simply be skipped. 

Going forward I can see this happening again, as new fields are added.

Reply:
@sqren I double checked with the elasticsearch team, and they said that the way I handled it is correct: 
```
boolean isFieldAvailable(def doc, def x) {
    return doc.containsKey(x) 
      && !doc[x].empty
  }
```
And the reason why it throws an error is that the user must define what will be the output when the field doesn't exist in the mapping.